### PR TITLE
feat: create user package and signer class

### PR DESCRIPTION
**Pull request by cmwaters**
Ref: https://github.com/celestiaorg/celestia-app/pull/2238

Creates a user package based from the the `KeyringSigner` in the blob module. A future PR will refactor the tests and remove the `KeyringSigner`

### CODE REVIEW

File: pkg/user/signer.go [hunk 1]
```diff
@@ -0,0 +1,375 @@
...
+		},
+	)
+	if err == nil {
+		return resp.TxResponse, err
+	}
+
+	// this is a bit brittle
+	if !strings.Contains(err.Error(), "not found") {
+		return &sdktypes.TxResponse{}, err
+	}
```
Comment (evan-forbes) [inline pkg/user/signer.go:223]: I've never understood why the response returned by the client here is nil when an error is returned! I wish it wasn't the case so we could avoid this

**Review by evan-forbes — APPROVED** [review]
:+1: LGTM

File: pkg/user/signer.go [hunk 2]
```diff
@@ -0,0 +1,375 @@
...
+	signature, _, err := s.keys.SignByAddress(s.address, bytesToSign)
+	if err != nil {
+		return nil, fmt.Errorf("error signing bytes: %w", err)
+	}
+
+	return signature, nil
+}
+
+// NewTxBuilder returns the default sdk Tx builder using the celestia-app encoding config
+func (s *Signer) txBuilder(opts ...TxOption) client.TxBuilder {
```
Comment (rootulp) [inline pkg/user/signer.go:341]: [not blocking] I think the godoc didn't get updated after a function rename

```suggestion
// txBuilder returns the default sdk Tx builder using the celestia-app encoding config
func (s *Signer) txBuilder(opts ...TxOption) client.TxBuilder {
```

**Review by rootulp — DISMISSED** [review]

File: pkg/user/signer.go [hunk 3]
```diff
@@ -0,0 +1,375 @@
...
+	txClient := tx.NewServiceClient(s.grpc)
+
+	resp, err := txClient.GetTx(
+		ctx,
+		&tx.GetTxRequest{
+			Hash: txHash,
+		},
+	)
+	if err == nil {
+		return resp.TxResponse, err
```
Comment (staheri14) [inline pkg/user/signer.go:217]: [nit] Just for improved readability: 
```suggestion
		return resp.TxResponse, nil
```

File: pkg/user/signer.go [hunk 4]
```diff
@@ -0,0 +1,375 @@
...
+func (s *Signer) ConfirmTx(ctx context.Context, txHash string) (*sdktypes.TxResponse, error) {
+	txClient := tx.NewServiceClient(s.grpc)
+
+	resp, err := txClient.GetTx(
+		ctx,
+		&tx.GetTxRequest{
+			Hash: txHash,
+		},
+	)
+	if err == nil {
```
Comment (staheri14) [inline pkg/user/signer.go:216]: [Question] Is this equality check correct? or should it be an inequality check?

File: pkg/user/signer.go [hunk 5]
```diff
@@ -0,0 +1,375 @@
...
+		case <-timer.C:
+			resp, err = txClient.GetTx(
+				ctx,
+				&tx.GetTxRequest{
+					Hash: txHash,
+				},
+			)
+
+			if err == nil {
+				return resp.TxResponse, err
```
Comment (staheri14) [inline pkg/user/signer.go:239]: [nit] For improved readability
```suggestion
				return resp.TxResponse, nil
```

File: pkg/user/signer.go [hunk 6]
```diff
@@ -0,0 +1,375 @@
...
+		},
+		Sequence: sequence,
+	}
+
+	err := builder.SetSignatures(draftsigV2)
+	if err != nil {
+		return fmt.Errorf("error setting draft signatures: %w", err)
+	}
+
+	// now we can use the data to produce the signature from each signer
```
Comment (staheri14) [inline pkg/user/signer.go:292]:
```suggestion
	// now we can use the data to produce the signature from the signer
```

**Review by staheri14 — APPROVED** [review]
LGTM!

File: pkg/user/signer.go (hunk 1 above)
Comment (cmwaters) [reply pkg/user/signer.go:223]: One day if we change it we're going to break the entire stack lol

File: pkg/user/signer.go (hunk 4 above)
Comment (cmwaters) [reply pkg/user/signer.go:216]: Yes just because the error handling happens later on

**Review by rootulp — APPROVED** [review]

**Review by staheri14 — APPROVED** [review]

**Review by evan-forbes — APPROVED** [review]

**Pull request merged**